python_modules/dagster/dagster/_core/definitions/partition.py

```diff
@@ -90,6 +94,10 @@ def __eq__(self, other: object) -> bool:
             return self.value == other.value and self.name == other.name
 
 
+class PendingPartition(Partition):
```

Comment:
I'm not fully grokking the need for this class - why not just construct a `Partition` object?

Reply:
we can, but it feels weird to create a `Partition` object where the key currently doesn't exist. If you're fine with that, I'm open to doing that too.

Reply:
That doesn't seem weird to me - an object is just an object. It doesn't have any context about the environment it's in.

Reply:
ok, got it. I'll remove this class